File: C++/fibonacci-heap.cpp
#include <iostream>
#include <vector>
#include <algorithm>

class FibonacciHeap {
private:
    int* minNode = nullptr;
    std::vector<int> nodes;
    
public:
    void insert(int value) {
        nodes.push_back(value);
        if (minNode == nullptr || value < *minNode) {
            minNode = &nodes.back();
        }
    }
    
    int getMin() {
        return *minNode;
    }
};

int main() {
    FibonacciHeap heap;
    heap.insert(5);
    heap.insert(2);
    heap.insert(8);
    std::cout << heap.getMin() << std::endl;  // 2
    
    return 0;
}

File: Python/fibonacci-heap.py
class FibonacciHeap:
    def __init__(self):
        self.min_node = None
        self.nodes = []
    
    def insert(self, value):
        self.nodes.append(value)
        if self.min_node is None or value < self.min_node:
            self.min_node = value

heap = FibonacciHeap()
heap.insert(5)
heap.insert(2)
heap.insert(8)
print(heap.min_node)  # 2
